renderdoc/driver/gl/wrappers/gl_emulated.cpp

```diff
@@ -2921,53 +2938,47 @@ void APIENTRY _glGetTexImage(GLenum target, GLint level, const GLenum format, co
     GL.glColorMask(GL_TRUE, GL_TRUE, GL_TRUE, GL_TRUE);
 
     GL.glActiveTexture(eGL_TEXTURE0);
-    GL.glBindTexture(eGL_TEXTURE_2D, boundTexture);
+    GL.glBindTexture(target, boundTexture);
 
     GLuint prog;
 
     {
       const char *vs =
-          "attribute vec2 pos;\n"
+          "#version 300 es\n"
+          "in vec2 pos;\n"
           "void main() { gl_Position = vec4(pos, 0.5, 0.5); }";
 
-      char fs_src[] =
-          "precision highp float;\n"
-          "uniform vec2 res;\n"
-          "uniform sampler2D srcTex;\n"
-          "void main() { gl_FragColor = texture2D(srcTex, vec2(gl_FragCoord.xy)/res).????; }";
-
-      char *swizzle = strchr(fs_src, '?');
+      const char *fs_sampler =
+          (target == GL_TEXTURE_2D)
+              ? "sampler2D"
+              : ((target == GL_TEXTURE_2D_ARRAY) ? "sampler2DArray" : "sampler3D");
+      const char *fs_dimension = (target == GL_TEXTURE_2D) ? "xy" : "xyz";
```

Comment:
I think it would be clearer to format this with if/else as you have with the swizzle below, as the multi-part trinary operator can be confusing to read.

Reply:
So, in new version it is only one trinary operator, so I left it.